controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -376,27 +366,24 @@ private String getTransactionName(String scope, String stream, long segmentId, U
     /**
      * This method sends a WireCommand to update table entries.
      *
-     * @param tableName           Qualified table name.
-     * @param entries             List of {@link TableEntry}s to be updated.
-     * @param delegationToken     The token to be presented to the segmentstore.
-     * @param clientRequestId     Request id.
-     * @return A CompletableFuture that, when completed normally, will contain the current versions of each {@link TableEntry}
+     * @param tableName       Qualified table name.
+     * @param entries         List of {@link TableSegmentEntry} instances to be updated.
+     * @param delegationToken The token to be presented to the Segment Store.
+     * @param clientRequestId Request id.
+     * @return A CompletableFuture that, when completed normally, will contain the current versions of each
+     * {@link TableSegmentEntry}.
      * If the operation failed, the future will be failed with the causing exception. If the exception can be retried
      * then the future will be failed with {@link WireCommandFailedException}.
      */
-    public CompletableFuture<List<KeyVersion>> updateTableEntries(final String tableName,
-                                                                  final List<TableEntry<byte[], byte[]>> entries,
-                                                                  String delegationToken,
-                                                                  final long clientRequestId) {
+    public CompletableFuture<List<TableSegmentKeyVersion>> updateTableEntries(final String tableName,
+                                                                              final List<TableSegmentEntry> entries,
+                                                                              String delegationToken,
+                                                                              final long clientRequestId) {
         final Controller.NodeUri uri = getTableUri(tableName);
         final WireCommandType type = WireCommandType.UPDATE_TABLE_ENTRIES;
-        List<ByteBuf> buffersToRelease = new ArrayList<>();
         List<Map.Entry<WireCommands.TableKey, WireCommands.TableValue>> wireCommandEntries = entries.stream().map(te -> {
             final WireCommands.TableKey key = convertToWireCommand(te.getKey());
-            ByteBuf valueBuffer = wrappedBuffer(te.getValue());
-            buffersToRelease.add(key.getData());
-            buffersToRelease.add(valueBuffer);
-            final WireCommands.TableValue value = new WireCommands.TableValue(valueBuffer);
```

Comment:
@andreipaduroiu Could you help undertstand how ByteBuf release is being handled here?

Reply:
I moved it up into `PravegaTablesStoreHelper`. It is the responsibility of whomever generates the `ByteBuf`s to release them. Now that they are no longer generated inside `SegmentHelper`, `PravegaTablesStoreHelper` is the one that both creates and releases them.

I needed to to this since the eventual `TableSegmentImpl` class deals with `ByteBuf`s that come from the eventual `KeyValueTableImpl` class, so I had to standardize the usage across both implementations.